### PR TITLE
Fix #8716 and fix #8717: Add missing extension span

### LINKED ISSUE #8716: dotr (REPL) 'extension method on' highlight crash
## Minimized example

Try to type in REPL:

```Scala
extension as on (s: String) { def expand = ??? }
```

## Output

Right after typing 'on':
```scala
extension as on
Trying to highlight erroneous position [14..16]. Input size: 15
Trying to highlight erroneous position [14..16]. Input size: 15
...
```

## Expectation

No highlight error

### LINKED ISSUE #8717: dotr (REPL) empty extension assertion error
## Minimized example

```Scala
scala> extension { }
```

## Output

```scala
Exception in thread "main" java.lang.AssertionError: assertion failed
	at dotty.DottyPredef$.assertFail(DottyPredef.scala:16)
	at dotty.tools.dotc.util.Spans$Span$.end$extension(Spans.scala:50)
	at dotty.tools.repl.ReplCompiler.compile(ReplCompiler.scala:165)
	at dotty.tools.repl.ReplDriver.compile(ReplDriver.scala:229)
	at dotty.tools.repl.ReplDriver.interpret(ReplDriver.scala:193)
	at dotty.tools.repl.ReplDriver.loop$1(ReplDriver.scala:127)
	at dotty.tools.repl.ReplDriver.runUntilQuit$$anonfun$1(ReplDriver.scala:130)
	at dotty.tools.repl.ReplDriver.withRedirectedOutput(ReplDriver.scala:148)
	at dotty.tools.repl.ReplDriver.runUntilQuit(ReplDriver.scala:130)
	at dotty.tools.repl.Main$.main(Main.scala:6)
	at dotty.tools.repl.Main.main(Main.scala)
```
(and REPL terminates)

## Expectation

Error similar to the one generated from compiler:
`anonymous instance must implement a type or have at least one extension method`
and REPL not terminating.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: tests/semanticdb/expect/Givens.expect.scala [hunk 1]
```diff
@@ -4,10 +4,10 @@ package b
...
+    de/*<-a::b::Givens.extension_sayGoodbye_B.*//*<-a::b::Givens.extension_sayGoodbye_B.sayGoodbye().[B]*//*<-a::b::Givens.extension_sayGoodbye_B.saySoLong().[B]*//*<-a::b::Givens.extension_sayGoodbye_B.sayGoodbye().(any)*//*<-a::b::Givens.extension_sayGoodbye_B.saySoLong().(any)*//*->a::b::Givens.extension_sayGoodbye_B.sayGoodbye().[B]*//*->a::b::Givens.extension_sayGoodbye_B.saySoLong().[B]*/f sayGoodbye/*<-a::b::Givens.extension_sayGoodbye_B.sayGoodbye().*/ = s"/*->scala::StringContext.apply().*/Goodbye, from $any/*->a::b::Givens.extension_sayGoodbye_B.sayGoodbye().(any)*/"/*->scala::StringContext#s().*/
     def saySoLong/*<-a::b::Givens.extension_sayGoodbye_B.saySoLong().*/ = s"/*->scala::StringContext.apply().*/So Long, from $any/*->a::b::Givens.extension_sayGoodbye_B.saySoLong().(any)*/"/*->scala::StringContext#s().*/
```
Comment (liufengyun) [inline tests/semanticdb/expect/Givens.expect.scala:11]: Are the changes here accidental?

Comment (nicolasstucki) [reply tests/semanticdb/expect/Givens.expect.scala:11]: Those positions where broken before and now are broken in a slightly different way.

Comment (nicolasstucki) [reply tests/semanticdb/expect/Givens.expect.scala:11]: Semanticdb must be looking at the wrong tree or some position gets badly propagated.

Comment (nicolasstucki) [reply tests/semanticdb/expect/Givens.expect.scala:11]: This is a separate issue

Comment (nicolasstucki) [reply tests/semanticdb/expect/Givens.expect.scala:11]: Opened https://github.com/lampepfl/dotty/issues/8725

**Pull request merged**